authorization.go

```diff
@@ -19,14 +19,17 @@ type Authorization struct {
 	Amount                     int64                   `json:"amount"`
 	Created                    json.Number             `json:"created"`
 	ReconciliationID           string                  `json:"reconciliation_id"`
-	PaymentMethod              PaymentMethodHref       `json:"payment_method"`
+	PaymentMethod              PaymentMethod           `json:"payment_method"`
 	ThreeDSecureAttributes     *ThreeDSecureAttributes `json:"three_d_secure_attributes"`
 	Installments               *Installments           `json:"installments"`
 	ProviderData               ProviderData            `json:"provider_data"`
-	ProviderSpecificData       map[string]interface{}  `json:"provider_specific_data"`
+	ProviderSpecificData       DecodedJSON             `json:"provider_specific_data"`
```

Comment:
Is there a reason to expect weird quoted jsons in this field?
PaymentsOS docs & bodybuilder suggest it should be an ordinary JSON object.

Reply:
You are right, but we can't control this field because a Bank fills and returns it.  I think that the bodybuilder doesn't have enough information about all cases for all providers.
DecodedJSON is a more unified decision here, it can check quoted data or not)